src/tests/ftest/io/llnl_mpi4py_hdf5.py

```diff
@@ -26,64 +26,29 @@
 import os
 import sys
 import json
-from avocado       import Test
-
-sys.path.append('./util')
-sys.path.append('../util')
-sys.path.append('../../../utils/py')
-sys.path.append('./../../utils/py')
-import AgentUtils
-import server_utils
+from apricot import TestWithServers
+
 import write_host_file
 from mpio_utils import MpioUtils, MpioFailed
-from daos_api import DaosContext, DaosPool, DaosApiError
+from daos_api import DaosPool, DaosApiError
 
-class LlnlMpi4py(Test):
+class LlnlMpi4pyHdf5(TestWithServers):
     """
-    Runs LLNL and MPI4PY test suites.
+    Runs LLNL, MPI4PY and HDF5 test suites.
+    :avocado: recursive
     """
 
     def setUp(self):
+        super(LlnlMpi4pyHdf5, self).setUp()
         # initialising variables
-        self.basepath = None
-        self.server_group = None
-        self.context = None
-        self.pool = None
         self.mpio = None
-        self.agent_sessions = None
-        self.hostlist_servers = None
-        self.hostfile_servers = None
         self.hostlist_clients = None
         self.hostfile_clients = None
-        self.createsvc = None
-
-        # get paths from the build_vars generated by build
-        with open('../../../.build_vars.json') as var_file:
-            build_paths = json.load(var_file)
-        self.basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
-
-        self.server_group = self.params.get("name", '/server_config/',
-                                            'daos_server')
-
-        # setup the DAOS python API
-        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
-
-        self.hostlist_servers = self.params.get("test_servers", '/run/hosts/')
-        self.hostfile_servers = write_host_file.write_host_file(
-            self.hostlist_servers, self.workdir)
-        print("Host file servers is: {}".format(self.hostfile_servers))
 
+        # setting client variables
```

Comment:
TestWithServers handles clients and servers, so lines 45-51 are no needed.

Reply:
For this test I need client hostile to have no "slots" hence mentioned it as "None". By default it is 1, hence I have to overwrite hostfile. I have removed self.hostlist_client lines though.

Reply:
Ah, sorry I missed that.  Maybe it would be worthwhile to have TestWithServers define variables for the slots that it passes into its write_host_file() calls so they could be overridden instead of creating a file twice.